// Compile via python -m grpc_tools.protoc -I. --python_out=. --grpc_python_out=. ServerRoomMusic.proto 

syntax = "proto3";

package ServerRoomMusic;

// Status values
enum Status {
  PENDING = 0;
  SUCCESS = 1;
  MATCH = 2;
  ERROR = 3;
}

// Kill Room
message KillRoomRequest {

}
message KillRoomResponse {

}

message JoinRoomRequest {
    string username = 1;
    string ClientAddress = 2;
}
message JoinRoomResponse {
    Status status = 1;
    string RoomTimeAddress = 2;
}

// Inform you left a room
message LeaveRoomRequest {
    string username = 1;
}
message LeaveRoomResponse {
    Status status = 1;
}

// Current State
message CurrentStateRequest {
}
message CurrentStateResponse {
    repeated string usernames = 1;
}

// Add Song
message AddSongRequest {
}
message AddSongResponse {
}

// Delete Song
message DeleteSongRequest {
}
message DeleteSongResponse {
}

// Start Song
message StartSongRequest {
}
message StartSongResponse {
}

// Pause Song
message PauseSongRequest {
}
message PauseSongResponse {
}

service ServerRoomMusic {
    rpc KillRoom(KillRoomRequest) returns (KillRoomResponse);
    rpc JoinRoom(JoinRoomRequest) returns (JoinRoomResponse);
    rpc LeaveRoom(LeaveRoomRequest) returns (LeaveRoomResponse);
    rpc CurrentState(CurrentStateRequest) returns (CurrentStateResponse);
    rpc AddSong(AddSongRequest) returns (AddSongResponse);
    rpc DeleteSong(DeleteSongRequest) returns (DeleteSongResponse);
    rpc StartSong(StartSongRequest) returns (StartSongResponse);
    rpc PauseSong(PauseSongRequest) returns (PauseSongResponse);
}